autoload/ale/fixers/xo.vim

```diff
@@ -1,23 +1,45 @@
 " Author: Albert Marquez - https://github.com/a-marquez
 " Description: Fixing files with XO.
 
-call ale#Set('javascript_xo_executable', 'xo')
-call ale#Set('javascript_xo_use_global', get(g:, 'ale_use_global_executables', 0))
-call ale#Set('javascript_xo_options', '')
+function! ale#fixers#xo#Fix(buffer) abort
+    let l:filetype = getbufvar(a:buffer, '&filetype')
+    let l:type = ''
+
+    if l:filetype =~# 'javascript'
+        let l:type = 'javascript'
+    elseif l:filetype =~# 'typescript'
+        let l:type = 'typescript'
+    endif
+
+    let l:executable = ale#handlers#xo#GetExecutable(a:buffer, l:type)
```

Comment:
Overall this PR looks really good. Tests are extensive, it uses version check for backward compatibility, and comments explaining why some things are done certain way. 

Just wondering if it is really necessary to pass l:type to GetExecutable(), GetOptions(), etc when this information can be easily extracted from the buffer argument that is also passed in to them.

Instead of passing l:type all over the place I think is better to simply use `getbufvar(a:buffer, '&filetype')` to get this information within those methods directly.

Reply:
Done.

Calculating `l:type` once and passing was a minor performance optimization. We can't rely on `getbufvar(a:buffer, '&filetype')` alone, though, given that the filetypes we're interested in are:

- `javascript`
- `javascriptreact`
- `typescript`
- `typescriptreact`

But we're limiting options to just `javascript_xo_...` and `typescript_xo_...`.